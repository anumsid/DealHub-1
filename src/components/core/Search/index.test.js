import React from 'react'
import { shallow } from 'enzyme'
import { stub } from 'sinon'
import toJson from 'enzyme-to-json'
import { SearchDeals } from './index'

describe('Search Component', () => {
  let wrapper
  let onClick
  beforeEach(() => {
    onClick = stub()
    const props = {
      onClick: onClick,
    }
    wrapper = shallow(<SearchDeals {...props} />)
  })

  it('Matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('Dispatches click when clicked', () => {
    wrapper.find('i').simulate('click')
    expect(onClick.calledOnce).toBe(true)
  })
})
