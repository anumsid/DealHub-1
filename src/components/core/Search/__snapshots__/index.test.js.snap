// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component Matches snapshot 1`] = `
<div
  className="searchInput ui small fluid icon input"
  style={
    Object {
      "width": "15rem",
    }
  }
>
  <input
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Search deals"
    type="text"
  />
  <i
    className="circular search link icon"
    onClick={[Function]}
  />
</div>
`;
